examples/dmd/heat_conduction_dmdc.cpp

```diff
@@ -0,0 +1,1028 @@
+                                       double kap, const Vector &u)
+    : TimeDependentOperator(f.GetTrueVSize(), 0.0), fespace(f), M(NULL), K(NULL),
+      T(NULL), current_dt(0.0),
+      M_solver(f.GetComm()), T_solver(f.GetComm()), z(height), b(height)
+{
+    const double rel_tol = 1e-8;
+
+    M = new ParBilinearForm(&fespace);
+    M->AddDomainIntegrator(new MassIntegrator());
+    M->Assemble(0); // keep sparsity pattern of M and K the same
+    M->FormSystemMatrix(ess_tdof_list, Mmat);
+
+    M_solver.iterative_mode = false;
+    M_solver.SetRelTol(rel_tol);
+    M_solver.SetAbsTol(0.0);
+    M_solver.SetMaxIter(100);
+    M_solver.SetPrintLevel(0);
+    M_prec.SetType(HypreSmoother::Jacobi);
+    M_solver.SetPreconditioner(M_prec);
+    M_solver.SetOperator(Mmat);
+
+    alpha = al;
+    kappa = kap;
+
+    T_solver.iterative_mode = false;
+    T_solver.SetRelTol(rel_tol);
+    T_solver.SetAbsTol(0.0);
+    T_solver.SetMaxIter(100);
+    T_solver.SetPrintLevel(0);
+    T_solver.SetPreconditioner(T_prec);
+
+    SetParameters(u);
+}
+
+void ConductionOperator::GetSource(Vector& s_v) const
+{
+    // Set grid function for f
+    ParGridFunction s_gf(&fespace);
+
+    FunctionCoefficient s(SourceFunction);
+    s.SetTime(GetTime());
+    s_gf.ProjectCoefficient(s);
+    s_gf.GetTrueDofs(s_v);
+}
+
+void ConductionOperator::Mult(const Vector &u, Vector &du_dt) const
+{
+    // Compute:
+    //    du_dt = M^{-1}*[f-K(u)]
+    // for du_dt
+
+    GetSource(b);
+    Kmat.Mult(u, z);
+    b.Add(-1.0, z);
+    M_solver.Mult(b, du_dt);
+}
+
+void ConductionOperator::ImplicitSolve(const double dt,
+                                       const Vector &u, Vector &du_dt)
+{
+    // Solve the equation:
+    //    du_dt = M^{-1}*[f-K(u + dt*du_dt)]
+    // for du_dt
+    if (!T)
+    {
+        T = Add(1.0, Mmat, dt, Kmat);
+        current_dt = dt;
+        T_solver.SetOperator(*T);
+    }
+    MFEM_VERIFY(dt == current_dt, ""); // SDIRK methods use the same dt
+
+    GetSource(b);
+    Kmat.Mult(u, z);
+    b.Add(-1.0, z);
+    T_solver.Mult(b, du_dt);
+}
+
+void ConductionOperator::SetParameters(const Vector &u)
+{
+    ParGridFunction u_alpha_gf(&fespace);
+    u_alpha_gf.SetFromTrueDofs(u);
+    for (int i = 0; i < u_alpha_gf.Size(); i++)
+    {
+        u_alpha_gf(i) = kappa + alpha*u_alpha_gf(i);
+    }
+
+    delete K;
+    K = new ParBilinearForm(&fespace);
+
+    GridFunctionCoefficient u_coeff(&u_alpha_gf);
+
+    K->AddDomainIntegrator(new DiffusionIntegrator(u_coeff));
+    K->Assemble(0); // keep sparsity pattern of M and K the same
+    K->FormSystemMatrix(ess_tdof_list, Kmat);
+    delete T;
+    T = NULL; // re-compute T on the next ImplicitSolve
+}
+
+ConductionOperator::~ConductionOperator()
+{
+    delete T;
+    delete M;
+    delete K;
+}
+
+double InitialTemperature(const Vector &x)
+{
+    return 1.0;
+}
+
+double SourceFunction(const Vector &x, const double t)
```

Comment:
We can make this source function continuously changing with respect to time variable, right? Not sure how DMDc will respond to this current step function in terms of time domain.

Reply:
Yes, I can make the source function separable type, i.e. $f(x,t) = g(x)h(t)$.

Reply:
Yes, I can make the source function become smooth.

Reply:
I think one of the parameters (`cx`, `cy`, or `amp`) needs to be time dependent?

Reply:
The `amp` is really the amplitude at t = 0. The time dependence is in line 1026, where the amplitude of the source function decays smoothly.